test/test_transforms_v2_refactored.py

```diff
@@ -2485,3 +2485,145 @@ def test_correctness(self):
                 assert isinstance(out_value, torch.Tensor) and not isinstance(out_value, datapoints.Datapoint)
             else:
                 assert isinstance(out_value, type(input_value))
+
+
+class TestErase:
+    INPUT_SIZE = (17, 11)
+    FUNCTIONAL_KWARGS = dict(
+        zip("ijhwv", [2, 2, 10, 8, torch.tensor(0.0, dtype=torch.float32, device="cpu").reshape(-1, 1, 1)])
+    )
+
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.uint8])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    def test_kernel_image(self, dtype, device):
+        check_kernel(F.erase_image, make_image(self.INPUT_SIZE, dtype=dtype, device=device), **self.FUNCTIONAL_KWARGS)
+
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.uint8])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    def test_kernel_image_inplace(self, dtype, device):
+        input = make_image(self.INPUT_SIZE, dtype=dtype, device=device)
+        input_version = input._version
+
+        output_out_of_place = F.erase_image(input, **self.FUNCTIONAL_KWARGS)
```

Comment:
We also need to check the Python objects, i,e. we need to add the following checks:

```py

        assert output_out_of_place is not input
...

        assert output_inplace is input

```

Reply:
I'll put them in for clarity, but I think these checks are redundant or potentially too strict:

- I'm not aware of a way to change the storage of an existing Python Tensor object. Meaning, it is safe to assume that if the storage pointer mismatches, we also have different Python objects
- I'm not sure if `inplace` refers to the Python object or the underlying storage. If it just applies to the storage `assert output_inplace is input` could actually fail:

    ```python
    import torch
    
    a = torch.rand(5)
    
    b = a.add_(1)
    
    assert b.data_ptr() == a.data_ptr()
    assert b is a
    
    c = b.view(-1)
    
    assert c.data_ptr() == a.data_ptr()
    assert c is not a
    ```

Reply:
> I'm not sure if inplace refers to the Python object or the underlying storage

Both. "inplace" means users are allowed to do

```py
F.erase_image(input, **self.FUNCTIONAL_KWARGS)
# use input now
```

without worrying about a return value. It's the same as for `inplace_()` operators. So it's best to enforce that contract in the tests as well.